### DISCUSSION TITLE
Is there a way to take the setting values I (un)checked and put them in the Infobar?

### DISCUSSION THREAD

**Discussion by TFWol** [gildas-lormeau/SingleFile#1531] [Q&A] [answered]
I'm trying to test saving a page with various combinations of settings and would like to have the current state of my settings available on hand in case it turns out the saved page was missing something I didn't catch until down the line.

The closest I can find are:

**Option: save the filename template data into the page**

> Check this option to save the template file data to the page. This allows you to recover template data previously used when saving the page and, for example, to ensure that the file is saved in the same directory as the original file when saving the page again.

**{profile-name}**

> the name of the profile used to save the page

The problem with just including the profile name is that I'll need to keep that profile around for reference, which means no renaming or modifying its settings. I'd rather not have dozens of profiles floating around confusing me.

I'm hoping maybe the first option somehow does this?
Where is says **"recover template data previously used when saving the page"**

**Comment by gildas-lormeau** [comment]
Sorry for the late reply, I could add variables returning the JSON content, and pretty-printed (e.g. tab + carriage returns) of all the options, e.g. respectively `{profile-data-json}` and `{profile-data}`. Would it suit your needs? I could also add a function to get a value given its key.

**Reply by TFWol** [reply]
Sorry, I've been distracted. 
That would be fine. Any method works as long as I can eyeball the settings (preferably in the same order as the options presented in Settings) 😄

You could even have an option to show the add-on's version (if there isn't something already) at the time of save so that it makes sense if a particular setting field was absent when comparing information.

**Reply by gildas-lormeau** [reply, marked as answer]
I added 2 variables `{options-json}` and `{options-text}`, and the function `%option-value<key>`. They won't be ordered like in the Settings page though unfortunately. I will add a variable for the version, this has already been suggested actually, see (4) here https://github.com/gildas-lormeau/SingleFile/issues/1504. The new variables and function will be available in the next version.

**Reply by TFWol** [reply]
Awesome, thanks!
